toplevel/vernac.mli

```diff
@@ -30,4 +30,4 @@ val process_expr : state:State.t -> Vernacexpr.vernac_control -> State.t
     echo the commands if [echo] is set. Callers are expected to handle
     and print errors in form of exceptions. *)
 val load_vernac : echo:bool -> check:bool -> interactive:bool ->
-  state:State.t -> string -> State.t
+  state:State.t -> ?ldir:Names.DirPath.t -> string -> State.t
```

Comment:
Tangential question: How does this work in OCaml?  Does OCaml automatically make this into an `option` type and insert `Some` when you pass it and `None` when you don't?

Reply:
Yes. But you have some extra syntactic sugar to set a default value, etc...

Reply:
Pretty much.  See the OCaml reference manual, section 4.1.1 _**Optional arguments**_.  Notice that if you pass `?ldir` a labelled parameter to further routines, as in `ccompile.ml`, you would generally use `~ldir`.  I don't use this often enough to get it right on the first compile.